ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,28 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner task.
+
+
+## Decision
```

Comment:
Can you please describe how secrets rotation/revocation will be handled?

Reply:
Proposal:
The user revokes the secret outside of Stonesoup.
The user generates a new credential outside of Stonesoup.
The user overwrites the secret in Stonesoup with a new one using our UI.

I propose that we don't have any more sophisticated secret rotation on our side at this point.

Reply:
Should we also support simply _deleting_ a secret in the UI?

Reply:
Is there any protection that Stonesoup could implement to invalidate/delete/etc. if secrets are referenced in tasks other than the intended target? This would have the potential to protect users from accidentally printed secrets as well as potentially "malicious" tasks that try to just get all secrets in the workspace.

I imagine this would be out of scope for an initial implementation, but it sounds like a good protective measure longer term.

Reply:
Thanks for the question, and I like @amfred 's proposal. That's a good question @arewm !

Reply:
Sure. Let me know if you want to chat about it.

Reply:
@sbose78 if the @amfred proposal is the accepted way, can you please add it to this ADR?

Reply:
https://github.com/redhat-appstudio/book/pull/64/commits/d1a2fd060af3e9317e98a03e76938c06d30f8b8b done!